src/main/java/org/jabref/gui/push/PushToEmacs.java

```diff
@@ -61,11 +63,12 @@ protected void initSettingsPanel() {
     }
 
     @Override
-    public void pushEntries(BibDatabase database, List<BibEntry> entries, String keys, MetaData metaData) {
+    public void pushEntries(BibDatabase database, List<BibEntry> entries, String keys, MetaData metaData, DialogService dialogService) {
```

Comment:
The `dialogService` makes more sense as a constructor argument, I think. Probably push it even to `AbstractPushToApplication`

Reply:
I thought so, too, but the concrete subclass of PushToEntries (operation) is just  passed as constructur argument as a dependency, so it is not possible

Reply:
Sorry, I don't understand what you mean. The push to application action only gets the `PushToApplicaiton` instance without knowing which concrete subclass it is. But this is not a problem as far as I can see since upon initialization each subclass would get a `DialogService` as a constructor argument.

Reply:
The problem is: `PushToApplicationAction` already gets an instance of the concrete push operation and only calls `pushEntries.
And these push-Operations are defined and instantiated here. I can't just pass the DialogService as Constructor arg.
The only thing I could do is to create an additional setter for the Dialog Service

https://github.com/JabRef/jabref/blob/fd98056c76bc340ab7ed62944e13414b20d5a117/src/main/java/org/jabref/gui/push/PushToApplications.java#L16-L20

Reply:
Why can't you pass the dialog service as a constructor argument to `PushToApplications`? It seems the only place where this class is initiated is https://github.com/JabRef/jabref/blob/fd98056c76bc340ab7ed62944e13414b20d5a117/src/main/java/org/jabref/gui/JabRefFrame.java#L580
where you have a dialog service at your disposal.

Reply:
done